snapcraft/projects.py

```diff
@@ -138,8 +138,16 @@ class App(ProjectModel):
     environment: Optional[Dict[str, str]]
     command_chain: List[str] = []
     sockets: Optional[Dict[str, Socket]]
+    adapter: Optional[str]  # no longer supported
     # TODO: implement passthrough (CRAFT-854)
 
+    @pydantic.validator("adapter")
+    @classmethod
+    def _validate_adapter(cls, name):
+        raise ValueError(
+            "The 'adapter' field is no longer supported, see the migration guide for details."
```

Comment:
We probably want to link to the DNXX we have and in there link to the migration guide, but in this case it is not a deprecation anymore https://github.com/snapcore/snapcraft/blob/main/snapcraft_legacy/internal/deprecations.py